src/content_script.js

```diff
@@ -32,50 +80,411 @@ function itemWidthCheck (target) {
+  // Detect if target is visible
+
+  const htmlBadgeDiv = createBadgeFragment(socialAction);
+
+  const htmlBadgeFragmentPromptButtonCancel = htmlBadgeDiv.querySelector(".fbc-badge-prompt-button-0");
+  const htmlBadgeFragmentPromptButtonAllow = htmlBadgeDiv.querySelector(".fbc-badge-prompt-button-1");
+  const htmlBadgeFragmentFenceDiv = htmlBadgeDiv.querySelector(".fbc-badge-fence");
+
+  htmlBadgeDiv.className = "fbc-badge " + badgeClassUId;
+
+  document.body.appendChild(htmlBadgeDiv);
+
+  const itemWidth = parseInt(target.offsetWidth, 10);
+  const itemHeight = parseInt(target.offsetHeight, 10);
+
+  const ratioCheck = (itemWidth / itemHeight);
+
+  let allowClickSwitch = false;
+
+  const badgeSmallSwitch = shouldBadgeBeSmall(ratioCheck, itemHeight);
+  if (badgeSmallSwitch) {
+    htmlBadgeDiv.classList.add("fbc-badge-small");
+  }
 
-  for (let querySelector of PATTERN_DETECTION_SELECTORS) {
+  positionFacebookBadge(target, badgeClassUId, itemWidth, badgeSmallSwitch);
+
+  // Show/hide prompt if login element
+  if (socialAction === "login"){
+    target.addEventListener("click", (e) => {
+      if (allowClickSwitch) {
+        // Button disabled. Either will trigger new HTTP request or page will refresh.
+        setTimeout(()=>{
+          location.reload(true);
+        }, 250);
+        return;
+      } else {
+        // Click badge, button disabled
+        e.preventDefault();
+        htmlBadgeFragmentFenceDiv.click();
+      }
+    });
+
+    htmlBadgeFragmentFenceDiv.addEventListener("click", (e) => {
+      e.preventDefault();
+      e.target.parentElement.classList.toggle("active");
+      positionPrompt( badgeClassUId );
+      target.classList.toggle("js-fbc-prompt-active");
+      document.body.classList.toggle("js-fbc-prompt-active");
+    });
+
+    // Add to Container "Allow"
+    htmlBadgeFragmentPromptButtonAllow.addEventListener("click", (e) => {
+      e.preventDefault();
+      allowClickSwitch = true;
+      browser.runtime.sendMessage("add-to-facebook-container");
+      target.click();
+    });
+
+    // Close prompt
+    htmlBadgeFragmentPromptButtonCancel.addEventListener("click", (e) => {
+      e.preventDefault();
+      document.body.classList.remove("js-fbc-prompt-active");
+      document.querySelector(".fbc-has-badge.js-fbc-prompt-active").classList.remove("js-fbc-prompt-active");
+      e.target.parentElement.parentNode.parentNode.classList.remove("active");
+    });
+  }
+}
+
+function findActivePrompt() {
+  const allBadges = document.querySelectorAll(".fbc-badge.active");
+  for (let badge of allBadges) {
+    return badge;
+  }
+}
+
+function closePrompt() {
+  const activePrompt = findActivePrompt();
+  activePrompt.classList.remove("active");
+  document.body.classList.remove("js-fbc-prompt-active");
+  document.querySelector(".fbc-has-badge.js-fbc-prompt-active").classList.remove("js-fbc-prompt-active");
+}
+
+function positionPrompt ( target ) {
```

Comment:
Change this param to something besides `target` - e.g., `targetSelector` (and drop the logic to prepend the `.` character). Better to not change the type (and meaning) of a function parameter inside the function.

Reply:
Understood. Revised!